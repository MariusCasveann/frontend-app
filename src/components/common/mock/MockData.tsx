import { GraphQLError } from 'graphql';
import { QueryAllProducts_allProducts } from './../../dashboard/atoms/ProductSelection/graphql/__generated__/QueryAllProducts';
import { QueryAllDomains_allDomains } from './../../dashboard/atoms/DomainSelection/graphql/__generated__/QueryAllDomains';
import { LightApplicationsOfProduct_lightApplicationsOfProduct } from './../../applications/graphql/__generated__/LightApplicationsOfProduct';
import { QueryBilling_billingQuery } from '../../dashboard/molecules/common/qraphql/__generated__/QueryBilling';
import {
    TeamByProductId_teamByProductId,
    TeamByProductId_teamByProductId_teamMembers
} from '../../teams/molecules/TeamMembers/graphql/__generated__/TeamByProductId';
import {
    QueryCurrentUser_currentUser,
    QueryCurrentUser_currentUser_roles
} from '../../authentication/molecules/UserProfile/graphql/__generated__/QueryCurrentUser';
import { SelectItemProps } from '../molecules/SearchableSelect/SearchableSelectPresenter';

export const mockGraphQLError: GraphQLError = {
    message: 'FATALITY',
    name: 'error',
    locations: undefined,
    path: undefined,
    nodes: undefined,
    positions: undefined,
    source: undefined,
    originalError: undefined,
    extensions: undefined
};

export const mockSelectedDomain: QueryAllDomains_allDomains = {
    id: 102,
    name: 'Engineering Platform',
    __typename: 'Domain'
};

export const mockSelectedProduct: QueryAllProducts_allProducts = {
    id: 50051,
    logo: null,
    name: 'a',
    __typename: 'LightProductDTO'
};

export const mockedApplications: LightApplicationsOfProduct_lightApplicationsOfProduct[] = [
    {
        id: 50051,
        staticPictureId: null,
        name: 'a',
        service: false,
        __typename: 'LightApplicationDTO'
    }
];

export const mockFunction = jest.fn();

export const mockDataEmpty: QueryBilling_billingQuery[] = [];

// tslint:disable-next-line:no-any
export const createMockedState = (loginAsProps?: Partial<{ data: any; loading: boolean; error: undefined }>): any => ({
    app: {
        authentication: {
            loginAs: {
                error: undefined,
                ...loginAsProps
            }
        },
        products: {
            selectedProduct: mockSelectedProduct
        }
    }
});

export const userProductMockedState = () => ({
    app: {
        authentication: {
            currentUser: mockCurrentUser
        },
        products: {
            selectedProduct: mockSelectedProduct
        }
    }
});

export const preSelectedProduct: QueryAllProducts_allProducts = {
    __typename: 'LightProductDTO',
    id: 'ALL',
    logo: 'ALL',
    name: 'All Products'
};

export const currentUserMockedState = () => ({
    app: {
        authentication: {
            currentUser: mockCurrentUser
        },
        products: {
            selectedProduct: preSelectedProduct
        }
    }
});

export const selectedProductMockedState = (
    loginAsProps?: Partial<{ data: any; loading: boolean; error: undefined }>
): any => ({
    app: {
        authentication: {
            loginAs: {
                error: undefined,
                ...loginAsProps
            }
        },
        products: {
            selectedProduct: mockSelectedProduct
        }
    }
});

export const buildDataExpectedEmptyResults = {
    labels: [],
    datasets: [
        {
            data: []
        }
    ]
};

export const mockErrorMessage = 'Error message';

export const mockTeamMember: TeamByProductId_teamByProductId_teamMembers = {
    __typename: 'User',
    email: 'test@mediamarktsaturn.com',
    firstName: 'Test',
    lastName: 'Test',
    id: '11',
    roles: [
        {
            name: 'WRITE_ALL',
            __typename: 'Role'
        },
        {
            name: 'a-admin',
            __typename: 'Role'
        }
    ]
};

export const mockTeamByProductId: TeamByProductId_teamByProductId = {
    __typename: 'Team',
    id: 2355,
    productOwner: {
        id: 11,
        firstName: 'TestUser',
        lastName: 'Test',
        __typename: 'User'
    },
    teamMembers: [
        {
            __typename: 'User',
            email: 'test1@mediamarktsaturn.com',
            firstName: 'Test',
            id: '1234',
            lastName: 'Test2',
            roles: [
                {
                    name: 'WRITE_ALL',
                    __typename: 'Role'
                },
                {
                    name: 'a-admin',
                    __typename: 'Role'
                }
            ]
        },
        {
            __typename: 'User',
            email: 'test2@mediamarktsaturn.com',
            firstName: 'Test2',
            id: '2234',
            lastName: 'Test2',
            roles: [
                {
                    name: 'WRITE_ALL',
                    __typename: 'Role'
                },
                {
                    name: 'a-admin',
                    __typename: 'Role'
                },
                {
                    name: 'test-admin',
                    __typename: 'Role'
                }
            ]
        }
    ]
};

export const mockCurrentUser: QueryCurrentUser_currentUser = {
    __typename: 'UserDTO',
    activated: true,
    createdBy: 'anonymousUser',
    createdDate: '2019-05-27T10:53:29.944312Z',
    email: 'test@mediamarktsaturn.com',
    firstName: 'Test',
    id: 49050,
    imageContent: null,
    langKey: null,
    lastModifiedBy: 'Test',
    lastModifiedDate: '2019-06-28T05:56:51.175813Z',
    lastName: 'Test',
    login: 'Test',
    roles: [
        {
            concretePermissions: [],
            id: 1300,
            name: 'DP God',
            readOnly: true,
            __typename: 'Role'
        },
        {
            concretePermissions: [],
            id: 4651,
            name: 'READ_ALL',
            readOnly: false,
            __typename: 'Role'
        },
        {
            concretePermissions: [],
            id: 69002,
            name: 'a-read',
            readOnly: true,
            __typename: 'Role'
        }
    ],
    staticPictureId: null,
    subscribedToNewsletter: true
};

export const mockEmptyTeam: TeamByProductId_teamByProductId = {
    __typename: 'Team',
    id: null,
    productOwner: null,
    teamMembers: []
};

export const mockCurrenUserRole: QueryCurrentUser_currentUser_roles = {
    concretePermissions: [],
    id: 1300,
    name: 'DP God',
    readOnly: true,
    __typename: 'Role'
};

export const mockForm = {
    getFieldDecorator: () => () => false,
    getFieldError: mockFunction,
    getFieldsError: mockFunction,
    getFieldsValue: mockFunction,
    getFieldValue: mockFunction,
    isFieldValidating: mockFunction,
    setFields: mockFunction,
    setFieldsValue: mockFunction,
    validateFields: mockFunction,
    validateFieldsAndScroll: mockFunction,
    isFieldsTouched: mockFunction,
    isFieldTouched: mockFunction,
    resetFields: mockFunction
};

export const mockHistory = {
    length: 7,
    push: mockFunction,
    action: 'PUSH',
    location: {
        hash: '',
        pathname: '/path',
        search: '',
        state: {}
    },
    replace: mockFunction,
    go: mockFunction,
    goBack: mockFunction,
    goForward: mockFunction,
    block: mockFunction,
    createHref: mockFunction,
    listen: mockFunction
};

export const mockSelectData: SelectItemProps[] = [
    {
        id: '351952',
        label: 'Test test'
    },
    {
        id: '80503',
        label: 'Test1 test1'
    },
    {
        id: '412605',
        label: 'Test2 test2'
    }
];
