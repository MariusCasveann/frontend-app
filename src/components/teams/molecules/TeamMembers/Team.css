.ant-btn-primary.add-member-button {
    float: right;
    background-color: var(--bo-color-interactive);
    border-color: none;
}

.ant-btn-primary.add-member-button:hover {
    background-color: var(--bo-color-hover);
}

.member-action-button button {
    color: var(--bo-color-interactive);
}

.member-action-button .ant-btn-link {
    border: none;
}

.member-action-button button:hover {
    color: var(--bo-color-hover);
}

.member-action-button .ant-btn-link.ant-btn-sm {
    padding: 0px;
}

.product-owner-name {
    padding-left: 30px;
    padding-right: 10px;
    color: var(--font-color-copy);
}

.product-owner-label {
    padding-bottom: 15px;
    font-weight: bold;
    color: var(--font-color-copy);
}

.product-owner-card {
    padding-top: 20px;
    padding-left: 15px;
    padding-bottom: 30px;
}

.card-title {
    color: var(--font-color-copy);
}

.edit-icon.anticon-edit.anticon {
    color: var(--bo-color-interactive);
}
