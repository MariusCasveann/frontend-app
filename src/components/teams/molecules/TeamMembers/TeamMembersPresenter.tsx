import React, { useState } from 'react';
import { Divider, Table, Avatar, Button } from 'antd';
import { ColumnProps } from 'antd/lib/table';
import LoadingOrErrorPresenter from '../../../common/organisms/LoadingOrError/LoadingOrErrorPresenter';
import { Card } from '../../../common/molecules/CardLayout/CardLayout';
import {
    TeamByProductId_teamByProductId,
    TeamByProductId_teamByProductId_teamMembers
} from './graphql/__generated__/TeamByProductId';
import {
    QueryCurrentUser_currentUser,
    QueryCurrentUser_currentUser_roles
} from '../../../authentication/molecules/UserProfile/graphql/__generated__/QueryCurrentUser';
import { QueryAllProducts_allProducts } from '../../../dashboard/atoms/ProductSelection/graphql/__generated__/QueryAllProducts';
import TeamProductOwnerPresenter from './TeamProductOwnerPresenter';
import { checkIfTeamHasData, checkIfMemberIsAdmin, checkIfCurrentUserIsAdmin } from '../../helpers/teamHelper';
import AdminRoleButton from '../../atoms/AdminRoleButton';
import ConfirmModal from '../../../common/molecules/Modals/ConfirmModal';
import '../../../common/molecules/CardLayout/CardLayout.css';
import './Team.css';
import CreateTeam from '../../../createTeam/CreateTeam';
import EditProductOwner from '../ProductOwner/EditProductOwner';
import AddTeamMembers from '../AddTeamMembers/AddTeamMembers';
import { History, LocationState } from 'history';

export interface TeamMemberProps {
    teamByProductId: TeamByProductId_teamByProductId;
    currentUser: QueryCurrentUser_currentUser | null;
    loading: boolean;
    error: string;
    selectedProduct: QueryAllProducts_allProducts;
    onAssign: (productId: string, userId: string) => void;
    onRevoke: (productId: string, userId: string) => void;
    onDelete: (productId: string, team: TeamByProductId_teamByProductId, userId: string) => void;
    history: History<LocationState>;
}

export const getColumns = (
    onAssign: (productId: string, userId: string) => void,
    onRevoke: (productId: string, userId: string) => void,
    selectedProduct: QueryAllProducts_allProducts,
    setUserId: (userId: string) => void,
    setVisible: (visbile: boolean) => void,
    isCurrentUserAdmin?: QueryCurrentUser_currentUser_roles | null
) => {
    const columns: Array<ColumnProps<TeamByProductId_teamByProductId_teamMembers | null>> = [
        {
            render: () => <Avatar size="small" icon="user" />,
            width: 20,
            key: 'avatar'
        },
        {
            title: 'Name',
            render: name => `${name.firstName} ${name.lastName}`,
            key: 'name'
        },
        {
            title: 'Email',
            dataIndex: 'email',
            key: 'email'
        },
        {
            title: 'Action',
            key: 'action',
            render: item => (
                <div className="member-action-button">
                    {isCurrentUserAdmin ? (
                        <>
                            <AdminRoleButton
                                onUpdateRole={() => {
                                    const admin = checkIfMemberIsAdmin(item, selectedProduct);
                                    admin
                                        ? onRevoke(selectedProduct.id, item.id)
                                        : onAssign(selectedProduct.id, item.id);
                                }}
                                isTeamMemberAdmin={checkIfMemberIsAdmin(item, selectedProduct)}
                            />
                            <Divider type="vertical" />
                        </>
                    ) : (
                        undefined
                    )}
                    <Button
                        type="link"
                        size="small"
                        onClick={() => {
                            setUserId(item.id);
                            setVisible(true);
                        }}
                    >
                        Delete
                    </Button>
                </div>
            )
        }
    ];
    return columns;
};

export default ({
    teamByProductId,
    loading,
    error,
    currentUser,
    selectedProduct,
    onAssign,
    onRevoke,
    onDelete,
    history
}: TeamMemberProps) => {
    const isCurrentUserAdmin = checkIfCurrentUserIsAdmin(currentUser, selectedProduct);
    const [visible, setVisible] = useState(false);
    const [userId, setUserId] = useState('');
    const [changeProductOwner, setChangeProductOwner] = useState<boolean>(false);
    const [addTeamMembers, setAddTeamMembers] = useState<boolean>(false);

    if (loading || error) {
        return (
            <Card width={100}>
                <LoadingOrErrorPresenter error={error} loading={loading} />
            </Card>
        );
    }
    if (checkIfTeamHasData(teamByProductId)) {
        return (
            <>
                <ConfirmModal
                    title="Delete"
                    visible={visible}
                    onCancel={() => setVisible(false)}
                    onOk={async () => {
                        await onDelete(selectedProduct.id, teamByProductId, userId);

                        setVisible(false);
                    }}
                    text="Do you want to delete this team member"
                />
                <Card title="Team members" width={100}>
                    <EditProductOwner
                        selectedProduct={selectedProduct}
                        teamByProductId={teamByProductId}
                        changeProductOwner={changeProductOwner}
                        setChangeProductOwner={setChangeProductOwner}
                    />
                    <AddTeamMembers
                        selectedProduct={selectedProduct}
                        teamByProductId={teamByProductId}
                        addTeamMembers={addTeamMembers}
                        setAddTeamMembers={setAddTeamMembers}
                    />
                    <TeamProductOwnerPresenter
                        teamByProductId={teamByProductId}
                        setChangeProductOwner={setChangeProductOwner}
                        setAddTeamMembers={setAddTeamMembers}
                    />
                    <Table
                        rowKey={record => (record && record.lastName) || ''}
                        columns={getColumns(
                            onAssign,
                            onRevoke,
                            selectedProduct,
                            setUserId,
                            setVisible,
                            isCurrentUserAdmin
                        )}
                        dataSource={(teamByProductId && teamByProductId.teamMembers) || []}
                        pagination={false}
                    />
                </Card>
            </>
        );
    }
    return <CreateTeam history={history} />;
};
