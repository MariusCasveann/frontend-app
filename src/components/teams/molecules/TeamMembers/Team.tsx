import React, { useEffect } from 'react';
import { connect } from 'react-redux';
import { loader } from 'graphql.macro';
import { compose, graphql } from 'react-apollo';
import { RootState } from '../../../../state/RootState';
import TeamMembersPresenter, { TeamMemberProps } from './TeamMembersPresenter';
import { QueryAllProducts_allProducts } from '../../../dashboard/atoms/ProductSelection/graphql/__generated__/QueryAllProducts';
import { QueryCurrentUser_currentUser } from '../../../authentication/molecules/UserProfile/graphql/__generated__/QueryCurrentUser';
import { getProductId } from '../../../common/helpers/productHelper';
import { useMutation } from '@apollo/react-hooks';
import { History, LocationState } from 'history';
import { QueryCurrentUser } from '../../../authentication/molecules/UserProfile/graphql/__generated__/QueryCurrentUser';
import { openSuccessNotification, openErrorNotification } from '../../../common/molecules/Notification/Notifications';
import {
    TeamByProductId_teamByProductId_teamMembers,
    TeamByProductId_teamByProductId
} from './graphql/__generated__/TeamByProductId';
import { teamMutationOptions, getNewTeamMembers } from '../../helpers/teamHelper';
import { successMessage, errorMessage } from '../../../../config/constants';

const queryTeamByProductId = loader('./graphql/TeamByProductId.graphql');
const mutationAssignAdminRole = loader('./graphql/TeamMemberAssignAdminRole.graphql');
const mutationRevokeAdminRole = loader('./graphql/TeamMemberRevokeAdminRole.graphql');
const mutationDeleteTeamMember = loader('./../../../createTeam/graphql/updateTeam.graphql');

interface PropsFromState {
    selectedProduct: QueryAllProducts_allProducts;
    currentUser: QueryCurrentUser_currentUser;
}

interface GraphQLProps {
    team: TeamMemberProps;
    currentUser: QueryCurrentUser;
    history: History<LocationState>;
}

export const mapStateToProps = (state: RootState) => ({
    selectedProduct: state.app.products.selectedProduct,
    currentUser: state.app.authentication.currentUser
});

export const handleDelete = async (
    teamByProductId: TeamByProductId_teamByProductId,
    userId: string,
    productId: string,
    deleteTeamMember: (variables: object) => void
) => {
    const newTeamMembers = getNewTeamMembers(teamByProductId, userId);

    try {
        await deleteTeamMember({
            variables: {
                productId,
                team: {
                    id: teamByProductId && teamByProductId.id,
                    productOwner: {
                        id: teamByProductId && teamByProductId.productOwner && teamByProductId.productOwner.id,
                        activated: true
                    },
                    teamMembers:
                        newTeamMembers &&
                        newTeamMembers.map((item: TeamByProductId_teamByProductId_teamMembers | null) => ({
                            id: (item && item.id) || '',
                            activated: true
                        }))
                }
            }
        });
        openSuccessNotification(successMessage, 'The team member is deleted successfully');
    } catch (error) {
        openErrorNotification(errorMessage, 'Deleting team member was not successful!');
    }
};

export const handleAssignAdminRole = async (
    productId: string,
    userId: string,
    assignAdminRole: (variables: object) => void
) => {
    try {
        await assignAdminRole({
            variables: {
                productId,
                userId
            }
        });
        openSuccessNotification(successMessage, 'Assign admin role was successful!');
    } catch (error) {
        openErrorNotification(errorMessage, 'Assign admin role was not successful!');
    }
};

export const handleRevokeAdminRole = async (
    productId: string,
    userId: string,
    revokeAdminRole: (variables: object) => void
) => {
    try {
        await revokeAdminRole({
            variables: {
                productId,
                userId
            }
        });
        openSuccessNotification(successMessage, 'Revoke admin role was successful!');
    } catch (error) {
        openErrorNotification(errorMessage, 'Revoke admin role was not successful!');
    }
};

export const Team = ({
    team: { teamByProductId, loading, error },
    currentUser,
    selectedProduct,
    history
}: PropsFromState & GraphQLProps) => {
    const [assignAdminRole] = useMutation(mutationAssignAdminRole, teamMutationOptions);
    const [revokeAdminRoleMutation] = useMutation(mutationRevokeAdminRole, teamMutationOptions);
    const [deleteTeamMember] = useMutation(mutationDeleteTeamMember, teamMutationOptions);

    useEffect(() => {
        if (teamByProductId === null) {
            history.push('/create-team');
        }
    }, [teamByProductId, history]);

    return (
        <TeamMembersPresenter
            onDelete={(productId = selectedProduct.id, team = teamByProductId, userId) => {
                handleDelete(team, userId, productId, deleteTeamMember);
            }}
            onAssign={async (productId = selectedProduct.id, userId) =>
                handleAssignAdminRole(productId, userId, assignAdminRole)
            }
            onRevoke={async (productId = selectedProduct.id, userId) =>
                handleRevokeAdminRole(productId, userId, revokeAdminRoleMutation)
            }
            selectedProduct={selectedProduct}
            loading={loading}
            error={error}
            history={history}
            teamByProductId={teamByProductId}
            currentUser={currentUser}
        />
    );
};

export default compose(
    connect(mapStateToProps),
    graphql(queryTeamByProductId, {
        name: 'team',
        options: ({ selectedProduct }: PropsFromState) => ({
            variables: {
                productId: getProductId(selectedProduct)
            },
            fetchPolicy: 'no-cache'
        })
    })
)(Team);
