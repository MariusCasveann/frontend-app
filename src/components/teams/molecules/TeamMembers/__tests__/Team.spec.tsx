import * as React from 'react';
import { Team, mapStateToProps, handleDelete } from '../Team';
import {
    mockSelectedProduct,
    selectedProductMockedState,
    mockTeamByProductId,
    mockCurrentUser,
    mockFunction
} from '../../../../common/mock/MockData';
import renderer from 'react-test-renderer';

const componentProps = {
    team: { teamByProductId: mockTeamByProductId, loading: false, error: 'error' },
    currentUser: mockCurrentUser,
    selectedProduct: mockSelectedProduct
};

describe('mapStateToProps', () => {
    it('should map state to props correctly', () => {
        const expectedResult = JSON.stringify({
            selectedProduct: mockSelectedProduct
        });
        const result = JSON.stringify(mapStateToProps(selectedProductMockedState()));
        expect(result).toEqual(expectedResult);
    });
});

describe('Team', () => {
    it('should return the expected output', () => {
        const wrapper = renderer.create(<Team {...componentProps} />).toJSON();
        expect(wrapper).toMatchSnapshot();
    });
});

describe('handleDelete', () => {
    it('should call handleDelete if teamByProductId exists', () => {
        const deleteTeamMember = mockFunction;
        handleDelete(mockTeamByProductId, '123', '12345', deleteTeamMember);
        expect(deleteTeamMember).toBeCalled();
    });
});

describe('handleRevokeAdminRole', () => {
    it('should call handleRevokeAdminRole if teamByProductId exists', () => {
        const revokeAdminRole = mockFunction;
        handleDelete(mockTeamByProductId, '123', '12345', revokeAdminRole);
        expect(revokeAdminRole).toBeCalled();
    });
});

describe('handleAssignAdminRole', () => {
    it('should call handleAssignAdminRole if teamByProductId exists', () => {
        const assignAdminRole = mockFunction;
        handleDelete(mockTeamByProductId, '123', '12345', assignAdminRole);
        expect(assignAdminRole).toBeCalled();
    });
});
/*import * as React from 'react';
import Team from '../Team';
import { MockedProvider } from 'react-apollo/test-utils';
import { mockTeamByProductId, mockCurrentUser } from '../../../../common/mock/MockData';
import * as renderer from 'react-test-renderer';
import TeamMembersPresenter from '../TeamMembersPresenter';
import { notification } from 'antd';
import { MockedResponse } from 'react-apollo/test-links';
import { Provider } from 'react-redux';
import createMockStore from 'redux-mock-store';
import * as apolloHooks from '@apollo/react-hooks';

let mocksDelete: ReadonlyArray<MockedResponse>;

const mockRootState = {
    app: {
        products: {
            selectedProduct: mockSelectedProduct
        }
    }
}

const deleteTeamMock = jest.fn().mockResolvedValue({});

describe('<Team/> delete team member mutation', () => {
    it('deleteTeamMember is called by TeamMembersPresenter and deletes team member', async () => {
        const success = jest.spyOn(notification, 'success')
        const useMutationSpied = jest.spyOn(apolloHooks, 'useMutation')
            .mockReturnValue([deleteTeamMock, { loading: false, called: false }])
        const root = renderer.create(
            <Provider store={createMockStore([])(mockRootState)}>
                <MockedProvider mocks={mocksDelete}>
                    <Team team={mockTeamByProductId} currentUser={mockCurrentUser} />
                </MockedProvider>
            </Provider>
        ).root;

        await root.findByType(TeamMembersPresenter).props.onDelete('50051', mockTeamByProductId, '3456')

        expect(success).toHaveBeenCalledTimes(1);
        useMutationSpied.mockRestore()
    });
});*/
