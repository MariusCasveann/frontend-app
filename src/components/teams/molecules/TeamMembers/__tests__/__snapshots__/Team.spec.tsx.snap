// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Team should return the expected output 1`] = `
<div
  className="card"
  style={
    Object {
      "flex": "1 1 100%",
    }
  }
>
  <div
    className="card-content-data"
  >
    <div
      className="card-content-wrapper"
    >
      <div>
        error
      </div>
    </div>
  </div>
</div>
`;
